lib/matplotlib/backend_managers.py

```diff
@@ -56,14 +56,13 @@ class ToolManager(object):
         `LockDraw` object to know if the message is available to write
     """
 
-    def __init__(self, canvas):
+    def __init__(self, manager):
```

Comment:
Not convinced yet, at the moment we can do that anyway, you created the example for #4143 which does exactly that, we have a toolbar in a different window, still very cool btw and shows the power and flexibility of MEP22 and MEP27 combined :D.

The only difference with this, as I see it, means we have two different tools, one that work on a set manager, and one that work on a set figure... and they can work unrelated, so you have tools that work on a specific manager unrelated to the main figure (and of course, you can have tools in the same toolbar pointing to different figures), in which case ``ToolManager.figure`` should really read ``ToolManager.default_figure`` as we can change the figure that individual tools work with later, as I see it (or do I go too far here)

However the tool selection mechanism will have to come from a specific FigureCanvas source, the keys come from the canvas, right?

I also wonder if this all becomes too complicated.  I don't know... will leave it here for now until I hear your response.